psi4/src/psi4/libmints/mintshelper.cc

```diff
@@ -3360,6 +3360,199 @@ std::vector<SharedMatrix> MintsHelper::ao_tei_deriv1(int atom, double omega,
     return grad;
 }
 
+/* 1st and 2nd derivatives of metric in AO basis  */
+
+std::vector<SharedMatrix> MintsHelper::ao_metric_deriv1(int atom, const std::string& aux_name) {
+    std::array<std::string, 3> cartcomp{ {"X", "Y", "Z"} };
+
+    auto aux = get_basisset(aux_name);
+    auto factory = std::make_shared<IntegralFactory>(aux, BasisSet::zero_ao_basis_set(), aux, BasisSet::zero_ao_basis_set());
+
+    auto ints = std::shared_ptr<TwoBodyAOInt>(factory->eri(1));
+    auto naux = aux->nbf();
+    int natom = basisset_->molecule()->natom();
+
+    std::vector<SharedMatrix> grad;
+    for (int p = 0; p < 3; p++) {
+        std::stringstream sstream;
+        sstream << "ao_metric_deriv1_" << atom << cartcomp[p];
+        grad.push_back(std::make_shared<Matrix>(sstream.str(), naux, naux));
+    }
+
+
+    const auto &buffers = ints->buffers();
+    for (int P = 0; P < aux->nshell(); P++) {
+        for (int Q = 0; Q < aux->nshell(); Q++) {
+            int Psize = aux->shell(P).nfunction();
+            int Qsize = aux->shell(Q).nfunction();
+
+            int Poff = aux->shell(P).function_index();
+            int Qoff = aux->shell(Q).function_index();
+
+            int Pcenter = aux->shell(P).ncenter();
+            int Qcenter = aux->shell(Q).ncenter();
+
+            size_t delta;
+
+            delta = 0L;
+
+            if (Pcenter != atom && Qcenter != atom) continue;
+
+            if (Pcenter == atom && Qcenter == atom) continue;
```

Comment:
A complete brainfart from me, but what's the reason for this check?  I can see that same-center terms with odd angular momentum would integrate to zero, but why the general case?

Reply:
The displacement doesn't cause the two orbitals to move relative to each other, because they're on the same atom. The integration is still over all space, so only the relative coordinates matter.